compose/service.py

```diff
@@ -701,6 +701,12 @@ def build(self, no_cache=False, pull=False, force_rm=False):
 
         build_opts = self.options.get('build', {})
         path = build_opts.get('context')
+        # If build argument is not defined and there is no environment variable
+        # with the same name then build argument value will be None
+        # Moreover it will be sent to the docker engine as None and then
+        # interpreted as string None which in many cases will fail the build
+        # That is why we filter out all pairs with value equal to None
+        buildargs = {k: v for k, v in build_opts.get('args', {}).items() if v != 'None'}
```

Comment:
I think this should be comparing against the literal value `None`, not the string `'None'`?